File: php/src/Models/EventBridgeConfigIn.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

class EventBridgeConfigIn implements \JsonSerializable
{
    private array $setFields = [];

    /**
     * @param string      $eventBusName The name or ARN of the event bus to receive the event
     * @param string|null $detailType   Free-form string, with a maximum of 128 characters
     * @param string|null $accessKeyId  Access key ID.
     *
     * Required (along with `secret_access_key`) if `role_arn` is blank.
     * @param string|null $secretAccessKey Secret access key.
     *
     * Required (along with `access_key_id`) if `role_arn` is blank.
     * @param string|null $roleArn    Role ARN for delegated authentication
     * @param string|null $externalId Shared secret passed as the STS ExternalId.
     *
     * Can only be set if `role_arn` is Some
     * @param string|null $region The region of the EventBridge bus.
     *
     * Currently a required field, but marked as optional because we may infer it from other fields in the future.
     */
    private function __construct(
        public readonly string $eventBusName,
        public readonly ?string $detailType = null,
        public readonly ?string $accessKeyId = null,
        public readonly ?string $secretAccessKey = null,
        public readonly ?string $roleArn = null,
        public readonly ?string $externalId = null,
        public readonly ?string $region = null,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of EventBridgeConfigIn with required fields.
     */
    public static function create(
        string $eventBusName,
    ): self {
        return new self(
            eventBusName: $eventBusName,
            detailType: null,
            accessKeyId: null,
            secretAccessKey: null,
            roleArn: null,
            externalId: null,
            region: null,
            setFields: ['eventBusName' => true]
        );
    }

    public function withDetailType(?string $detailType): self
    {
        $setFields = $this->setFields;
        $setFields['detailType'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $detailType,
            accessKeyId: $this->accessKeyId,
            secretAccessKey: $this->secretAccessKey,
            roleArn: $this->roleArn,
            externalId: $this->externalId,
            region: $this->region,
            setFields: $setFields
        );
    }

    public function withAccessKeyId(?string $accessKeyId): self
    {
        $setFields = $this->setFields;
        $setFields['accessKeyId'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $this->detailType,
            accessKeyId: $accessKeyId,
            secretAccessKey: $this->secretAccessKey,
            roleArn: $this->roleArn,
            externalId: $this->externalId,
            region: $this->region,
            setFields: $setFields
        );
    }

    public function withSecretAccessKey(?string $secretAccessKey): self
    {
        $setFields = $this->setFields;
        $setFields['secretAccessKey'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $this->detailType,
            accessKeyId: $this->accessKeyId,
            secretAccessKey: $secretAccessKey,
            roleArn: $this->roleArn,
            externalId: $this->externalId,
            region: $this->region,
            setFields: $setFields
        );
    }

    public function withRoleArn(?string $roleArn): self
    {
        $setFields = $this->setFields;
        $setFields['roleArn'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $this->detailType,
            accessKeyId: $this->accessKeyId,
            secretAccessKey: $this->secretAccessKey,
            roleArn: $roleArn,
            externalId: $this->externalId,
            region: $this->region,
            setFields: $setFields
        );
    }

    public function withExternalId(?string $externalId): self
    {
        $setFields = $this->setFields;
        $setFields['externalId'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $this->detailType,
            accessKeyId: $this->accessKeyId,
            secretAccessKey: $this->secretAccessKey,
            roleArn: $this->roleArn,
            externalId: $externalId,
            region: $this->region,
            setFields: $setFields
        );
    }

    public function withRegion(?string $region): self
    {
        $setFields = $this->setFields;
        $setFields['region'] = true;

        return new self(
            eventBusName: $this->eventBusName,
            detailType: $this->detailType,
            accessKeyId: $this->accessKeyId,
            secretAccessKey: $this->secretAccessKey,
            roleArn: $this->roleArn,
            externalId: $this->externalId,
            region: $region,
            setFields: $setFields
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
            'eventBusName' => $this->eventBusName];

        if (null !== $this->detailType) {
            $data['detailType'] = $this->detailType;
        }
        if (isset($this->setFields['accessKeyId'])) {
            $data['accessKeyId'] = $this->accessKeyId;
        }
        if (isset($this->setFields['secretAccessKey'])) {
            $data['secretAccessKey'] = $this->secretAccessKey;
        }
        if (isset($this->setFields['roleArn'])) {
            $data['roleArn'] = $this->roleArn;
        }
        if (isset($this->setFields['externalId'])) {
            $data['externalId'] = $this->externalId;
        }
        if (isset($this->setFields['region'])) {
            $data['region'] = $this->region;
        }

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            eventBusName: \Svix\Utils::deserializeString($data, 'eventBusName', true, 'EventBridgeConfigIn'),
            detailType: \Svix\Utils::deserializeString($data, 'detailType', false, 'EventBridgeConfigIn'),
            accessKeyId: \Svix\Utils::deserializeString($data, 'accessKeyId', false, 'EventBridgeConfigIn'),
            secretAccessKey: \Svix\Utils::deserializeString($data, 'secretAccessKey', false, 'EventBridgeConfigIn'),
            roleArn: \Svix\Utils::deserializeString($data, 'roleArn', false, 'EventBridgeConfigIn'),
            externalId: \Svix\Utils::deserializeString($data, 'externalId', false, 'EventBridgeConfigIn'),
            region: \Svix\Utils::deserializeString($data, 'region', false, 'EventBridgeConfigIn')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}

File: java/lib/src/main/java/com/svix/models/EventBridgeConfigIn.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.svix.Utils;

import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(getterVisibility = Visibility.NONE, setterVisibility = Visibility.NONE)
public class EventBridgeConfigIn {
    @JsonProperty private String eventBusName;
    @JsonProperty private String detailType;
    @JsonProperty private String accessKeyId;
    @JsonProperty private String secretAccessKey;
    @JsonProperty private String roleArn;
    @JsonProperty private String externalId;
    @JsonProperty private String region;

    public EventBridgeConfigIn() {}

    public EventBridgeConfigIn eventBusName(String eventBusName) {
        this.eventBusName = eventBusName;
        return this;
    }

    /**
     * The name or ARN of the event bus to receive the event
     *
     * @return eventBusName
     */
    @javax.annotation.Nonnull
    public String getEventBusName() {
        return eventBusName;
    }

    public void setEventBusName(String eventBusName) {
        this.eventBusName = eventBusName;
    }

    public EventBridgeConfigIn detailType(String detailType) {
        this.detailType = detailType;
        return this;
    }

    /**
     * Free-form string, with a maximum of 128 characters
     *
     * @return detailType
     */
    @javax.annotation.Nullable
    public String getDetailType() {
        return detailType;
    }

    public void setDetailType(String detailType) {
        this.detailType = detailType;
    }

    public EventBridgeConfigIn accessKeyId(String accessKeyId) {
        this.accessKeyId = accessKeyId;
        return this;
    }

    /**
     * Access key ID.
     *
     * <p>Required (along with `secret_access_key`) if `role_arn` is blank.
     *
     * @return accessKeyId
     */
    @javax.annotation.Nullable
    public String getAccessKeyId() {
        return accessKeyId;
    }

    public void setAccessKeyId(String accessKeyId) {
        this.accessKeyId = accessKeyId;
    }

    public EventBridgeConfigIn secretAccessKey(String secretAccessKey) {
        this.secretAccessKey = secretAccessKey;
        return this;
    }

    /**
     * Secret access key.
     *
     * <p>Required (along with `access_key_id`) if `role_arn` is blank.
     *
     * @return secretAccessKey
     */
    @javax.annotation.Nullable
    public String getSecretAccessKey() {
        return secretAccessKey;
    }

    public void setSecretAccessKey(String secretAccessKey) {
        this.secretAccessKey = secretAccessKey;
    }

    public EventBridgeConfigIn roleArn(String roleArn) {
        this.roleArn = roleArn;
        return this;
    }

    /**
     * Role ARN for delegated authentication
     *
     * @return roleArn
     */
    @javax.annotation.Nullable
    public String getRoleArn() {
        return roleArn;
    }

    public void setRoleArn(String roleArn) {
        this.roleArn = roleArn;
    }

    public EventBridgeConfigIn externalId(String externalId) {
        this.externalId = externalId;
        return this;
    }

    /**
     * Shared secret passed as the STS ExternalId.
     *
     * <p>Can only be set if `role_arn` is Some
     *
     * @return externalId
     */
    @javax.annotation.Nullable
    public String getExternalId() {
        return externalId;
    }

    public void setExternalId(String externalId) {
        this.externalId = externalId;
    }

    public EventBridgeConfigIn region(String region) {
        this.region = region;
        return this;
    }

    /**
     * The region of the EventBridge bus.
     *
     * <p>Currently a required field, but marked as optional because we may infer it from other
     * fields in the future.
     *
     * @return region
     */
    @javax.annotation.Nullable
    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    /**
     * Create an instance of EventBridgeConfigIn given an JSON string
     *
     * @param jsonString JSON string
     * @return An instance of EventBridgeConfigIn
     * @throws JsonProcessingException if the JSON string is invalid with respect to
     *     EventBridgeConfigIn
     */
    public static EventBridgeConfigIn fromJson(String jsonString) throws JsonProcessingException {
        return Utils.getObjectMapper().readValue(jsonString, EventBridgeConfigIn.class);
    }

    /**
     * Convert an instance of EventBridgeConfigIn to an JSON string
     *
     * @return JSON string
     */
    public String toJson() throws JsonProcessingException {
        return Utils.getObjectMapper().writeValueAsString(this);
    }
}
